webapp/controller/ProductDetails.controller.js

```diff
@@ -1,47 +1,98 @@
 sap.ui.define([
 	"sap/ui/core/mvc/Controller",
-	"sap/ui/model/json/JSONModel"
-], function (Controller, JSONModel) {
+	"sap/ui/model/json/JSONModel",
+	"sap/m/MessageToast",
+	"sap/m/MessageBox",
+	"sap/ui/core/format/DateFormat",
+	"sap/ui/core/Core",
+	"sap/ui/model/ValidateException",
+	"sap/base/util/deepEqual"
+], function (Controller, JSONModel, MessageToast, MessageBox, DateFormat, Core, ValidateException, deepEqual) {
 	"use strict";
 
 	return Controller.extend("products.app.controller.ProductDetails", {
 		
         onInit: function() {
-            const oModel = new JSONModel();
+			
             const oComponent = this.getOwnerComponent();
             const oRouter = oComponent.getRouter();
-            
-            oModel.loadData("./model/products.json", "", false, "", true);
-			oModel.loadData("./model/categories.json", "", false, "", true);
-			oModel.loadData("./model/suppliers.json", "", false, "", true);
-
-            this.getView().setModel(oModel);
+            const oAvailabilityModel = new JSONModel({
+				Availability: [
+					{
+						key: "true",
+						value: this.getTextFromI18n("InStockText")
+					},
+					{
+						key: "false",
+						value: this.getTextFromI18n("OutOfStockText")
+					}
+				]
+			})
+			Core.getMessageManager().registerObject(this.getView(), true);
```

Comment:
I believe when you provide "handleValidation" property in manifest.json you don't have explicitly use registerObject like that. Try to remove it and see whether validation works or not